streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala

```diff
@@ -102,6 +100,180 @@ class WriteAheadLogSuite extends SparkFunSuite with BeforeAndAfter {
     }
   }
 
+  test("WriteAheadLogUtils - wrap WriteAheadLog in BatchedWriteAheadLog when batching is enabled") {
+    def getBatchedSparkConf: SparkConf =
+      new SparkConf().set("spark.streaming.driver.writeAheadLog.allowBatching", "true")
+
+    val emptyConf = getBatchedSparkConf
+    assertDriverLogClass[FileBasedWriteAheadLog](emptyConf, isBatched = true)
+    assertReceiverLogClass[FileBasedWriteAheadLog](emptyConf)
+
+    // Verify setting driver WAL class
+    val conf1 = getBatchedSparkConf.set("spark.streaming.driver.writeAheadLog.class",
+      classOf[MockWriteAheadLog0].getName())
+    assertDriverLogClass[MockWriteAheadLog0](conf1, isBatched = true)
+    assertReceiverLogClass[FileBasedWriteAheadLog](conf1)
+
+    // Verify receivers are not wrapped
+    val receiverWALConf = getBatchedSparkConf.set("spark.streaming.receiver.writeAheadLog.class",
+      classOf[MockWriteAheadLog0].getName())
+    assertDriverLogClass[FileBasedWriteAheadLog](receiverWALConf, isBatched = true)
+    assertReceiverLogClass[MockWriteAheadLog0](receiverWALConf)
+  }
+}
+
+/** Common tests for WriteAheadLogs that we would like to test with different configurations. */
+abstract class CommonWriteAheadLogTests(
+    allowBatching: Boolean,
+    closeFileAfterWrite: Boolean,
+    testTag: String = "")
+  extends SparkFunSuite with BeforeAndAfter {
+
+  import WriteAheadLogSuite._
+
+  val hadoopConf = new Configuration()
+  var tempDir: File = null
+  var testDir: String = null
+  var testFile: String = null
+  var writeAheadLog: WriteAheadLog = null
+  protected def testPrefix = if (testTag != "") testTag + " - " else testTag
+
+  before {
+    tempDir = Utils.createTempDir()
+    testDir = tempDir.toString
+    testFile = new File(tempDir, "testFile").toString
+    if (writeAheadLog != null) {
+      writeAheadLog.close()
+      writeAheadLog = null
+    }
+  }
+
+  after {
+    Utils.deleteRecursively(tempDir)
+  }
+
+  test(testPrefix + "read rotating logs") {
```

Comment:
offline discussion, call this `readAll`, as all subclasses may not be *rotating* logs.